src/plugins/intel_cpu/src/nodes/rnn.cpp

```diff
@@ -274,8 +275,57 @@ bool RNN::isSupportedOperation(const std::shared_ptr<const ov::Node>& op, std::s
     return true;
 }
 
+namespace {
+class RnnShapeInfer : public NgraphShapeInfer {
+public:
+    RnnShapeInfer(std::shared_ptr<ov::Node> op) :
+        NgraphShapeInfer(make_shape_inference(op), EMPTY_PORT_MASK) {
+            is_sequence = one_of(op->get_type_info(),
+                ov::op::v5::GRUSequence::get_type_info_static(),
+                ov::op::v0::LSTMSequence::get_type_info_static(),
+                ov::op::v5::LSTMSequence::get_type_info_static(),
+                ov::op::v5::RNNSequence::get_type_info_static(),
+                ov::op::internal::AUGRUSequence::get_type_info_static());
+
+            if (is_sequence) {
+                native_order = false;
+                const auto& rtInfo = op->get_rt_info();
+                if (rtInfo.count("seqAxis")) {
+                    native_order = rtInfo.at("seqAxis").as<int64_t>() == 0;
```

Comment:
I would say shapeInfer() override is still looks better than `is_sequence` and "native _order" initialization code copy-paste. 
Can we at least extract initialization code into separate functions and reuse with RNNNode ctor?